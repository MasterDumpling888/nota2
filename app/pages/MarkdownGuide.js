import React from 'react';
import { ScrollView, View, Text, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import PageBox from '../components/PageBox';
import Markdown from 'react-native-markdown-display';

const markdownGuideContent = `
# Markdown Guide

## Headers
# H1
## H2
### H3
#### H4
##### H5
###### H6

## Emphasis
*Italic*  
**Bold**  
***Bold Italic***

## Lists
### Unordered
- Item 1
- Item 2
  - Subitem 1
  - Subitem 2

### Ordered
1. Item 1
2. Item 2
   1. Subitem 1
   2. Subitem 2

## Links
[Google](https://www.google.com)

## Images
![Alt text](https://via.placeholder.com/150)

## Blockquotes
> This is a blockquote.

## Code
\`Inline code\`

\`\`\`
Block code
\`\`\`

## Tables
| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |
| Cell 3   | Cell 4   |
`;

function MarkdownGuide({ navigation }) {
  return (
    <SafeAreaView style={{ flex: 1 }}>
      <PageBox title="Markdown Guide" onClose={() => navigation.goBack()}>
        <ScrollView style={styles.container}>
          <Markdown>{markdownGuideContent}</Markdown>
        </ScrollView>
      </PageBox>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    padding: 16,
  },
});

export default MarkdownGuide;
