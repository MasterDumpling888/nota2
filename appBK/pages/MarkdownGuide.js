import React from 'react';
import { ScrollView, View, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import PageBox from '../components/PageBox';
import MarkdownCard from '../components/MarkdownCard';

const markdownGuides = [
  {
    title: 'Headers',
    markdown: `
# H1
## H2
### H3
#### H4
##### H5
###### H6
    `,
  },
  {
    title: 'Emphasis & Strong',
    markdown: `
*Italic*
**Bold**
***Bold Italic***
`,
  },
  {
    title: 'Lists',
    markdown: `
### Unordered
- Item 1
- Item 2
  - Subitem 1
  - Subitem 2
### Ordered
1. Item 1
2. Item 2
   1. Subitem 1
   2. Subitem 2
`,
  },
  {
    title: 'Links',
    markdown: `
[Google](https://www.google.com)
    `,
  },
  {
    title: 'Images',
    markdown: `
![Alt text](https://via.placeholder.com/150)
    `,
  },
  {
    title: 'Blockquotes',
    markdown: `
> This is a blockquote.
    `,
  },
  {
    title: 'Code',
    markdown: `
\`Inline code\`

\`\`\`
Block code
\`\`\`
    `,
  },
  {
    title: 'Tables',
    markdown: `
| Month    | Savings |
|----------|---------|
| January  | $250    |
| February | $80     |
| March    | $420    |
    `,
  },
];

function MarkdownGuide({ navigation }) {
  return (
    <SafeAreaView style={styles.container}>
      <PageBox title="Markdown Guide" onClose={() => navigation.goBack()}>
        <ScrollView>
          {markdownGuides.map((guide, index) => (
            <MarkdownCard key={index} title={guide.title} markdown={guide.markdown} />
          ))}
        </ScrollView>
      </PageBox>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#181818',
    alignItems: 'center',
  },
});

export default MarkdownGuide;
