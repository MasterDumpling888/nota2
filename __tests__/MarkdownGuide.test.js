import React from 'react';
import { render, waitFor } from '@testing-library/react-native';
import MarkdownGuide from '../app/pages/MarkdownGuide';


// Mock navigation
const mockNavigation = {
    goBack: jest.fn(),
    navigate: jest.fn(),
};

// Mock PageBox component
jest.mock('../app/components/PageBox', () => (props) => (
    <mock-PageBox>{props.children}</mock-PageBox>
));

// Mock MarkdownCard component
jest.mock('../app/components/MarkdownCard', () => (props) => (
    <mock-MarkdownCard>{props.title}</mock-MarkdownCard>
));


test('MarkdownGuide snapshot test', () => {
    const { toJSON } = render(<MarkdownGuide navigation={mockNavigation} />);
    expect(toJSON()).toMatchSnapshot();
});