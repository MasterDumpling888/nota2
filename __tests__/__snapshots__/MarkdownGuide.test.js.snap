// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownGuide snapshot test 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#181818",
      "flex": 1,
    }
  }
>
  <mock-PageBox>
    <RCTScrollView>
      <View>
        <mock-MarkdownCard>
          Headers
        </mock-MarkdownCard>
        <mock-MarkdownCard>
          Emphasis & Strong
        </mock-MarkdownCard>
        <mock-MarkdownCard>
          Lists
        </mock-MarkdownCard>
        <mock-MarkdownCard>
          Links
        </mock-MarkdownCard>
        <mock-MarkdownCard>
          Images
        </mock-MarkdownCard>
        <mock-MarkdownCard>
          Blockquotes
        </mock-MarkdownCard>
        <mock-MarkdownCard>
          Code
        </mock-MarkdownCard>
        <mock-MarkdownCard>
          Tables
        </mock-MarkdownCard>
      </View>
    </RCTScrollView>
  </mock-PageBox>
</RNCSafeAreaView>
`;
